### PR TITLE
[Bug]: Unable to export translated values from the grid

### LINKED ISSUE #12959: [Bug]: Unable to export translated values from the grid
### Expected behavior

Exported xlxs/csv file has translated values

### Actual behavior

Exported xlxs/csv file has values equal to the key, not translation.

image
image
image
image

### Steps to reproduce

1. open object grid options
2. select desired attirbutes, that have translations in messages/admin translations
3. select english language
4. observe the result on grid
5. export grid

additional info:
these may be linked to this issue:
https://github.com/pimcore/pimcore/issues/12958
https://github.com/pimcore/pimcore/issues/12903

### PR DESCRIPTION
**Pull request by robertSt7**
resolve #12959

### CODE REVIEW

File: models/DataObject/Service.php [hunk 1]
```diff
@@ -1890,6 +1892,11 @@ public static function getCsvData($requestedLanguage, LocaleServiceInterface $lo
                 }
 
                 $rowData = self::getCsvDataForObject($object, $requestedLanguage, $fields, $helperDefinitions, $localeService, false, $context);
+                foreach ($rowData as &$fieldData) {
+                    if($translator !== null) {
+                        $fieldData = $translator->trans($fieldData, [], 'admin', $requestedLanguage);
+                    }
+                }
```
Comment (mcop1) [inline models/DataObject/Service.php:1899]:
```suggestion
                if($translator !== null) {
                    foreach ($rowData as &$fieldData) {
                        $fieldData = $translator->trans($fieldData, [], 'admin', $requestedLanguage);
                    }
                }
```

We don´t have to do the foreach if there is no translator, right?

**Comment by brusch** [comment]
See https://github.com/pimcore/pimcore/issues/12959#issuecomment-1255083803

**Comment by brusch** [comment]
Let's keep this open for discussion for now ...

**Comment by robertSt7** [comment]
The PR is closed due to inactivity.

**Pull request closed without merging**